src/games/supported/Alien.cpp

```diff
@@ -142,3 +146,42 @@ void AlienSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+//Returns a list of mode that the game can be played in.
+ModeVect AlienSettings::getAvailableModes(){
+    ModeVect modes(4);
+    for(unsigned i = 0; i < 4; i++){
```

Comment:
C++11 could indeed simplify some bits of the code at various places, but was not enabled at the time this PR was written. If the decision is made to move to this standard, I can rewrite the parts were it makes really sense to use new features.
As a side note, some parts of the code use auto_ptr, which has been deprecated. It could be a good opportunity to get rid of them as well